scripts/bootstrap.sh.in

```diff
@@ -88,56 +88,52 @@ run_kubelet_start() {
         ${SALT_CALL} --local --retcode-passthrough state.apply metalk8s.kubeadm.init.kubelet-start saltenv=metalk8s-@VERSION@ pillarenv=metalk8s-@VERSION@
 }
 
-install_salt_master() {
-        # TODO: Create pod using kubelet with master image
-        ${YUM} install -y salt-master
+configure_salt() {
+    ${SALT_CALL} --retcode-passthrough state.apply metalk8s.salt.minion.configured saltenv=metalk8s-@VERSION@ pillarenv=metalk8s-@VERSION@
+    rm -f "${SALT_MINION_FILE_CLIENT_LOCAL_CONF}"
 }
 
-configure_salt() {
-        # TODO: Use a salt state and maybe other stuff
-        cat > "${SALT_MINION_FILE_CONF}" << EOF
-master: localhost
+deploy_salt_master() {
+    # inject salt master image into containerd cache
+    # gunzip /srv/scality/metalk8s-dev/images/salt-master-2018.3.3-1.tar.gz -c >/tmp/salt-master-2018.3.3-1.tar
+    # ctr cri load /tmp/salt-master-2018.3.3-1.tar
 
-# use new module.run format
-use_superseded:
-  - module.run
-EOF
-        rm "${SALT_MINION_FILE_CLIENT_LOCAL_CONF}"
-        cat > "${SALT_MASTER_FILE_CONF}" << EOF
-file_roots:
-  metalk8s-@VERSION@:
-    - /srv/scality/metalk8s-dev/salt
-pillar_roots:
-  metalk8s-@VERSION@:
-    - /srv/scality/metalk8s-dev/pillar
-peer:
-  .*:
-    - x509.sign_remote_certificate
-EOF
-        ${SYSTEMCTL} restart salt-master.service
-        ${SYSTEMCTL} enable salt-master.service
-        ${SYSTEMCTL} restart salt-minion.service
-        ${SYSTEMCTL} enable salt-minion.service
-        sleep 20
-        salt-key -A -y
+    # Run deployment
+    ${SALT_CALL} --retcode-passthrough state.apply metalk8s.salt.master saltenv=metalk8s-@VERSION@ pillarenv=metalk8s-@VERSION@
+    # Allow some time for the image to be pulled instead of looping later
+    sleep 5
```

Comment:
IMO if it's really needed that should be part of the state

Reply:
Yeah, not sure what this does TBH...

Reply:
#656